bin/dlang/make

```diff
@@ -7,28 +7,42 @@
 # Build D applications
 set -eu
 
-# Set default arguments if none are passed
-if test $# -eq 0
-then
-    set -- all test
-fi
-
 # Paths
-r=$(dirname $0)/../..
-beaver=$r/bin/beaver
+r="$(readlink -f $(dirname $0)/../..)"
+beaver="$r/bin/beaver"
 
 # Use general dlang utilities
 . $r/lib/dlang.sh
 
 # Set the DC and DVER environment variables and export them to docker
 set_dc_dver
-export BEAVER_DOCKER_VARS="${BEAVER_DOCKER_VARS:-} DC DVER D2_ONLY"
+export BEAVER_DOCKER_VARS="${BEAVER_DOCKER_VARS:-} DMD DC DVER D2_ONLY F V"
+
+# We have arguments, forward to make and exit
+if test $# -gt 0
+then
+    # Just forward the make call
+    exec "$beaver" make "$@"
+fi
+
+# No argumets, run a default build
 
 # First convert code if we are building a D2 version of a D1/2 project
 if test "$D2_ONLY" = false -a "$DVER" -eq 2
 then
-    $beaver make d2conv
+    "$beaver" make d2conv
 fi
 
-# Then just build
-$beaver make "$@"
+# Build default target
+"$beaver" make all
+
+# Test and upload coverage reports
+test_cov()
+{
+    what="$1"
+    "$beaver" make "$what"
+    "$beaver" dlang codecov -n "$what" -F "$what"
+    rm -f .*.lst
+}
+test_cov unittest
+test_cov integrationtest
```

Comment:
Until we have a way to output coverage reports from different integration tests to different folders and submit them all, I would recommend disabling this part completely. Right now you will get misleading flakey reports from whatever integration tests has run last.

Reply:
OK, good point.

Reply:
https://github.com/sociomantic-tsunami/makd/issues/107

Reply:
Updated.